torch/csrc/distributed/c10d/symm_mem/nvshmem_extension.cuh

```diff
@@ -15,6 +16,8 @@ void initialize_nvshmem_with_store(
 // operations.
 TORCH_API void nvshmemx_cumodule_init(uintptr_t module);
 
+TORCH_API void nvshmem_put(at::Tensor& inp, c10::intrusive_ptr<c10d::symmetric_memory::SymmetricMemory> dest, const int64_t peer);
```

Comment:
You can use
`using c10d::symmetric_memory::SymmetricMemory`
to make the type shorter.